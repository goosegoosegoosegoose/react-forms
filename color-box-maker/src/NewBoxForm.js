import React, { useState } from "react";

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
        bgcolor: "",
        width: "",
        height: ""
    };

    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }))
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox(formData.bgcolor, formData.width, formData.height);
        setFormData(INITIAL_STATE);
    };

    return (
        <form onSubmit = {handleSubmit}>
            <label htmlFor="bgcolor">Background Color: </label>
            <input 
                id="bgcolor"
                placeholder="background color"
                type="text"
                name="bgcolor"
                value={formData.bgcolor}
                onChange={handleChange}
            /><br/>

            <label htmlFor="width">Width: </label>
            <input
                id="width"
                placeholder="width"
                type="number"
                name="width"
                value={formData.width}
                onChange={handleChange}
            /><br/>

            <label htmlFor="height">Height: </label>
            <input
                id="height"
                placeholder="height"
                type="number"
                name="height"
                value={formData.height}
                onChange={handleChange}
            /><br/>
            
            <button>Submit</button>
        </form>
    )
}

export default NewBoxForm;