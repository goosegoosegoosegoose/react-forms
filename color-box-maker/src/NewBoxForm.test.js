import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm"

it("smoke test", () => {
    render(<NewBoxForm />);
});

it("snapshot test", () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});