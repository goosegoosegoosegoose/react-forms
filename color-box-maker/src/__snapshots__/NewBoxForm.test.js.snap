// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <form>
    <label
      for="bgcolor"
    >
      Background Color: 
    </label>
    <input
      id="bgcolor"
      name="bgcolor"
      placeholder="background color"
      type="text"
      value=""
    />
    <br />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      name="width"
      placeholder="width"
      type="number"
      value=""
    />
    <br />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      name="height"
      placeholder="height"
      type="number"
      value=""
    />
    <br />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
